### DISCUSSION TITLE
Check database schema compared with latest migration

### DISCUSSION THREAD

**Discussion by stunox** [prisma/prisma#18138] [Q&A] [answered]
When I start my server, and the db is not up to date compared with the migration, it doesn't send any error.

I want to know if there is a way to c**heck if the database schema is same as the migration**, (in the _code_ or in the _shell_, or **both** ?)

Thank you

**Comment by nurul3101** [comment, marked as answer]
Hello @stunox 👋 

You can use prisma [migrate diff](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff) command to check the difference between your database and the migration history.

Here's the command you can use:
```cmd
npx prisma migrate diff \
 --from-url "$DATABASE_URL" \
 --to-migrations ./migrations \
 --script > script.sql
```

If there is no sql query logged in `script.sql` then it would mean that migrations and the database are in sync.

**Reply by nurul3101** [reply]
@stunox If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by oom-** [reply]
@nurul3101 it's not working anymore, 
I tried `npx prisma migrate diff --to-migrations .\prisma\migrations\ --script --from-config-datasource .\prisma.config.ts`
but getting : 
```
Error: You must pass the `--shadow-database-url` flag or set `datasource.shadowDatabaseUrl` in your `prisma.config.ts` if you want to diff a migrations directory.
```
I do not have a `shadow-database`, I only have one DB running locally in a docker container (btw setting the `shadowDatabaseUrl` to my local database is returning `Error: The shadow database you configured appears to be the same as the main database. Please specify another shadow database.`)
